product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -29,6 +29,8 @@ def price_compute(
         if price_type == 'partner':
             partner = self.env.context.get('partner_id', False) or \
                 self.env.context.get('partner', False)
+            if partner and not isinstance(partner, (int,)):
```

Comment:
Instead of looking for not being integer, better to check if is `BaseModel`

Reply:
done!